Lib/uuid.py

```diff
@@ -355,7 +370,7 @@ def _find_mac(command, args, hw_identifiers, get_index):
                         try:
                             word = words[get_index(i)]
                             mac = int(word.replace(b':', b''), 16)
-                            if mac:
+                            if ~(mac & (1<<41)):
```

Comment:
I don't understand the usage of the ~int operator. 

What do you want to test? If a bit is unset? Do you mean "if not(mac & (1<<41)):" or "if (mac & (1<<41)) == 0:" ?

Please write a private subfunction for this test.

Reply:
Yeah, that was completely botched.  Now that I've had some sleep <wink> I've fixed this with a helper function.

Reply:
Also with the latest commit, we keep track of the first locally administered MAC found.  If no universally administered MAC is found, we return the first local MAC.

And the way the getters work, if there are no local or universal MACs found, then `getnode()` ultimately falls back to the RFC 4122 specified random MAC, with the multicast bit set.  So it should never be the case that `getnode()` doesn't return an RFC-defined node address.